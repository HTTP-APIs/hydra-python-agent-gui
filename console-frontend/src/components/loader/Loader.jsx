import React from 'react'
import { withStyles } from '@material-ui/styles';
import logo from '../../assets/images/hydra_eco_logo.png'

const styles = theme => ({
    image: {
        position: "absolute",
        top: "50%",
        left: "50%",
        width: "120px",
        height: "120px",
        margin: `-60px 0 0 -60px`,
        animation:`spin 4s linear infinite`,
        borderRadius: `30%`
    },
    "@keyframes spin" : { 
        "100%" : { 
            transform: "rotate(360deg)" 
        } 
    }
});



const Loader = (props) => {
    const classes = props.classes
    return(
        <div className={classes.loader}>
            <img className={classes.image} src={logo} alt="Loader Image"  />
        </div>
    )
}


export default withStyles(styles)(Loader);