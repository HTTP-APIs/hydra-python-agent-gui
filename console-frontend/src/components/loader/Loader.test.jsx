import React from 'react';
import { shallow, mount, render} from 'enzyme';
import Loader from './Loader';
import renderer from 'react-test-renderer';
import logo from '../../assets/images/hydra_eco_logo.png';


describe('<Loader />', () => {
    it('snapshot render', function() {
        const component = renderer.create(<Loader />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('loader image', function() {
        const wrapper = mount(<Loader />);
        const image = wrapper.find('div img')
        expect(image).toBeTruthy();
        expect(image.length).toEqual(1);
    });
})