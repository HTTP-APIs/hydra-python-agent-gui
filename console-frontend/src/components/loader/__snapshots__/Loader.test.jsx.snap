// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> snapshot render 1`] = `
<div>
  <img
    alt="Loader Image"
    className="Loader-image-1"
    src="hydra_eco_logo.png"
  />
</div>
`;
